### DISCUSSION TITLE
CME Futures & Trailing stop order

### DISCUSSION THREAD

**Discussion by shial4** [stensoosaar/IBKit#10] [Q&A]
Hi! 
General question for IB and its functionality, 
I want to trade futures. Still waiting for them to approve me to be able to do it via IB. I used to use Tradovate before. 

Do you know how one would make market order with trailing stop & optional take profit stop

[this article](https://www.interactivebrokers.com/en/trading/orders/trailing-stops.php) says it is not possible for futures. Is it true? 
Do you have knowledge about it? 
With Tradovate & CME I was able to use stop loss & take profit and trailing trailing limit orders with CME futures & EUREX 

I have this code, with the  order `. trail` that I've added in last PR (the active one)
```
public func makeOrder(contract: IBContract, action: IBAction, totalQuantity: Double) throws {
        guard let account = identifiers.first else { return }
        let requestID = client.nextRequestID
        try client.placeOrder(
            requestID,
            contract: contract,
            order: .trail(
                trailingPercent: 0.15,
                action: action,
                quantity: 1,
                account: account
            )
        )
    }
```

would you be able testing it in paper trading for me? much appreciate help. Thanks in advance.

**Comment by stensoosaar** [comment]
To be honest, never used them, so I had to find out (used MYM)
If your primary order is limit or stoplimit order, you can add use trailing order  

I'll keep the data flowing and let my own strategy to decide entry/exit points. Gives you a chance to evaluate importance of each spike...

**Reply by shial4** [reply]
can you elaborate? 
I use trailing stop orders to make myself secure, just in case, something happens like internet breaks down. or gateway crashes or my program fails. 

You know things happens :P 
and trailing limit order gives you this buffer, to have your own exit Strat. while still being secured

**Reply by stensoosaar** [reply]
Nothing wrong with that. Everyone has their own rules.
I also use protective stop for extra insurance but using support/resistance levels to determine action areas to evaluate the situation.

**Reply by shial4** [reply]
If I didn't find your lib so far in, I probably wouldn't use IB as they seem so out of date haha
would you mind sharing some examples of how you place order with stop limit?

**Reply by stensoosaar** [reply]
IB as a broker or data provider is fine if you dont count time in milliseconds. But if you want to deal with multiple assets or multiple timeframes, your own object store and db are must.

**Comment by shial4** [comment]
Ok, closing this due to the more DM approach now.
